anvio/scgsdatabase.py

```diff
@@ -0,0 +1,791 @@
+                pickle.dump(dictionnary__low_ident, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+
+    def is_database_exists(self):
+        if os.path.exists(os.path.join(self.SCG_data_dir, 'SCG-A.hmm.gz')):
+            raise ConfigError("It seems you already have SCG database installed in '%s', please use --reset flag if you want to re-download it." % self.SCG_data_dir)
+
+
+    def get_remote_version(self):
+        content = read_remote_file(self.database_url + '/VERSION')
+
+
+    def getfasta(self,pathfa):
+        fasta = u.ReadFasta(pathfa)
+        return(fasta)
+
+    def domatrix(self,taxofile):
+        """format tsv in dictonnary with code as key and liste for taxonomy for value"""
+        with open(taxofile,'r') as taxo:
+            linestaxo=taxo.readlines()
+            listspecies=[]
+            matrix={}
+            for line in linestaxo:
+                names=line.split("\t")
+                code=str(names[0])
+                taxos=names[1].split(";")
+                """for name in taxos[0:-2]:
+                    if not re.match('[a-z]__[A-Z]{1}[a-z]{1,}',name):"""
+
+                taxo= re.sub(r'_[A-Z]*$', '', taxos[-1]).rstrip()
+                leveltaxo3 = re.sub(r'(\\*[a-z])\_[A-Z]', r'\1', taxo)
+                last_taxo1 = re.sub(r'sp[0-9]{1,}$', '', leveltaxo3)
+                last_taxo=last_taxo1.rstrip(" ").replace(" ","_")
+                if not re.match('s__[A-Z]{1}[a-z]{1,}',last_taxo):
+                    continue
+                taxos[-1]=last_taxo
+
+                matrix[code]=taxos
+            return(matrix)
+
+
+
+    def creatsubfa(self,matrix_taxonomy,listerefence,pathfile,keylevel,dictionnary__pickel_taxo,dictionnary_level=None):
+
+        listtaxo=[]
+        unmatch=[]
+        newfasta=""
+        with open(self.outtsv,'a') as tsv:
+            with open(pathfile,'a') as file:
+                for refence in listerefence:
+                    pathfa=os.path.join(self.outputdirectory, "diamonddb",refence+".faa")
+                    run.info("pathfa",pathfa)
+                    fasta=u.ReadFasta(pathfa)
+                    if dictionnary_level!=None:
+                        for code, taxonomy in matrix_taxonomy.items():
+                            index,name=self.match(fasta,keylevel,taxonomy,code,listtaxo)
+                            if not index:
+                                unmatch.append(name)
+                                continue
+                            else:
+                                listtaxo.append(name)
+                                newfasta=newfasta+">"+fasta.ids[index]+"\n"+fasta.sequences[index]+"\n"
+                                tsv.write(fasta.ids[index]+"\t"+';'.join([matrix_taxonomy[fasta.ids[index]]])+"\n")
+                                dictionnary__pickel_taxo[fasta.ids[index]]=matrix_taxonomy[fasta.ids[index]]
+                    else:
+                        i=0
```

Comment:
you often use `i`s or `j`s for loops. not very phytonic. why not `for fasta_id in fasta.ids: (...)`?

it would make the code much more readable. please revisit all your loops and change this coding style.